mne/evoked.py

```diff
@@ -996,22 +996,33 @@ def grand_average(all_evoked, interpolate_bads='eeg'):
     return grand_average
 
 
-def merge_evoked(all_evoked):
-    """Merge/concat evoked data
+def combine_evoked(all_evoked, weights='nave'):
+    """Merge evoked data by weighted addition
 
     Data should have the same channels and the same time instants.
+    Subtraction can be performed by passing negative weights (e.g., [1, -1]).
 
     Parameters
     ----------
     all_evoked : list of Evoked
-        The evoked datasets
+        The evoked datasets.
+    weights : list of float | str
```

Comment:
There is something a little bit weird about using `[1] * len(all_evoked)`, though. The `all_nave` comes out as:
```
all_nave = sum(w * w / e.nave for w, e in zip(weights, all_evoked))
evoked.data = sum(...) / all_nave
```
So if we have e.g. two datasets with 200 averages that we want to add together with equal weighting, then we'll get:
```
all_nave = 1.*1./200 + 1.*1./200  # 1 / 100
evoked.data = sum(...) * 100
```
Instead of using all ones, we could use something like the geometric or arithmetic mean of the `nave`s of the evokeds, but I don't think it's going to look too good either way.

@agramfort is using the squared weight divided by the `nave` the best way to recompute the resulting `nave`? You originally had something even more complicated, which I think was the reciprocal of that value:

https://github.com/mne-tools/mne-python/pull/461/files#diff-738430c6d6954d2d42b2632b6fd1a0bdR631

I'm starting to wonder whether your solution was a better way to go...